### DISCUSSION TITLE
How can Nuxt better integrate typescript? It's troublesome to import every time.

### DISCUSSION THREAD

**Discussion by bbhxwl** [nuxt/nuxt#29546] [Questions]
There is also no code prompt for typescript, which is very troublesome to use. I have to manually reference the entire path of the type in order to use a type.

demo
https://github.com/bbhxwl/testNuxt3

<img width="1209" alt="image" src="https://github.com/user-attachments/assets/17d96194-b407-419c-a0a4-0b1fa37c378b">
<img width="1097" alt="image" src="https://github.com/user-attachments/assets/322a962a-d356-4d40-a38e-057de35a1e6f">
<img width="1133" alt="image" src="https://github.com/user-attachments/assets/2f5147a3-70b6-4982-b78b-1527d372563d">

**Comment by Abolfazl2049** [comment]
you can add the entire folder path to nuxt.config.ts in autoImport section. more info here https://nuxt.com/docs/guide/concepts/auto-imports

**Reply by bbhxwl** [reply]
> you can add the entire folder path to nuxt.config.ts in autoImport section. more info here https://nuxt.com/docs/guide/concepts/auto-imports

Isn't this only able to automatically import Vue components?

**Reply by Abolfazl2049** [reply]
as i said you can add the types folder path inside your nuxt.config.ts file , something like this :
`export default defineNuxtConfig({
  imports: {
    dirs: ["composables/**", "types/**"],
  },
})`
and in your ts files in types folder export types like this : 
`interface test {
  name: string;
  id: number;
}
export type { test };
`
and you can use it in your app without import like this :
`let test = (test: test) => {};`
